e2e/_suites/fleet/integrations.go

```diff
@@ -17,18 +18,74 @@ const elasticEnpointIntegrationTitle = "Endpoint Security"
 // IntegrationPackage used to share information about a integration
 type IntegrationPackage struct {
 	packageConfigID string          `json:"packageConfigId"`
-	name            string          `json:"name"`
-	title           string          `json:"title"`
-	version         string          `json:"version"`
+	Name            string          `json:"name"`
+	Title           string          `json:"title"`
+	Version         string          `json:"version"`
 	json            *gabs.Container // json representation of the integration
 }
 
+type Policy struct {
+	ID            string             `json:"id,omitempty"`
+	Name          string             `json:"name"`
+	Description   string             `json:"description"`
+	Namespace     string             `json:"namespace"`
+	Enabled       bool               `json:"enabled"`
+	AgentPolicyID string             `json:"policy_id"`
+	OutputID      string             `json:"output_id"`
+	Inputs        []PolicyInput      `json:"inputs"`
+	Package       IntegrationPackage `json:"package"`
+}
+
+type PolicyInput struct {
+	Type    string         `json:"type"`
+	Enabled bool           `json:"enabled"`
+	Streams []interface{}  `json:"streams"`
+	Vars    map[string]Var `json:"vars,omitempty"`
+}
+
+type Var struct {
+	Value interface{} `json:"value"`
+	Type  string      `json:"type"`
+}
+
 // addIntegrationToPolicy sends a POST request to Fleet adding an integration to a configuration
 func addIntegrationToPolicy(integrationPackage IntegrationPackage, policyID string) (string, error) {
-	name := integrationPackage.name + "-test-name"
-	description := integrationPackage.title + "-test-description"
 
-	body, err := kibanaClient.AddIntegrationToPolicy(integrationPackage.name, name, integrationPackage.title, description, integrationPackage.version, policyID)
+	policy := Policy{
+		AgentPolicyID: policyID,
+		Name:          integrationPackage.Name + "-test-" + "-" + policyID,
+		Description:   integrationPackage.Title + "-test-description",
+		Namespace:     "default",
+		Enabled:       true,
+		Package:       integrationPackage,
```

Comment:
This will fix the `null` error while posting the endpoint integration. If you take a look at https://github.com/elastic/e2e-testing/pull/924/files#diff-738d8d068f7c4134a147cd0f4b65a3cc7045391b344ee47e9a17722705b76749L70, you'll find that the previous value was an empty array.
```suggestion
		Package:       integrationPackage,
		Inputs:        []PolicyInput{},
```

Reply:
Good catch, thanks. I had already most of suggestions implemented, which I managed to leave out thanks to my clever misuse of `git add` 🤦 - but this one I missed.